test/suites/alpha/drift/suite_test.go

```diff
@@ -99,6 +100,35 @@ var _ = Describe("Drift", Label("AWS"), func() {
 		nodeTemplate.Spec.AMIFamily = &v1alpha1.AMIFamilyCustom
 		nodeTemplate.Spec.AMISelector = map[string]string{"aws-ids": oldCustomAMI}
 		nodeTemplate.Spec.UserData = awssdk.String(fmt.Sprintf("#!/bin/bash\n/etc/eks/bootstrap.sh '%s'", env.ClusterName))
+		provisioner.Spec.Requirements = append(provisioner.Spec.Requirements, v1.NodeSelectorRequirement{Key: v1.LabelArchStable, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.ArchitectureAmd64, v1alpha5.ArchitectureArm64}})
+
+		env.ExpectCreated(pod, nodeTemplate, provisioner)
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+
+		machine := env.EventuallyExpectCreatedMachineCount("==", 1)[0]
+		node := env.EventuallyExpectNodeCount("==", 1)[0]
+		nodeTemplate.Spec.AMISelector = map[string]string{"aws-ids": customAMI}
+		env.ExpectCreatedOrUpdated(nodeTemplate)
+
+		Eventually(func(g Gomega) {
+			g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(machine), machine)).To(Succeed())
+			g.Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted)).ToNot(BeNil())
```

Comment:
We can drop this because the `IsTrue` automatically catches the nil status condition